### PR TITLE
Fix fixture manager vendor/plugin edge case

### PR DESCRIPTION
**Pull request by jadb**
Refs #5579

### CODE REVIEW

**Comment by markstory** [comment]
[Tests are failing](https://travis-ci.org/cakephp/cakephp/jobs/46077823)

**Comment by markstory** [comment]
Other than the test failures, this seems reasonable.

File: tests/TestCase/TestSuite/FixtureManagerTest.php [hunk 1]
```diff
@@ -72,4 +72,23 @@ public function testFixturizePlugin()
...
+    {
+        $test = $this->getMock('Cake\TestSuite\TestCase');
+        $test->fixtures = ['plugin.Company/TestPluginThree.articles'];
+        $this->manager->fixturize($test);
+        $fixtures = $this->manager->loaded();
+        $this->assertCount(1, $fixtures);
+        $this->assertArrayHasKey('plugin.Company/TestPluginThree.articles', $fixtures);
+        $this->assertInstanceOf(
+            'Company\TestPluginThree\Test\Fixture\ArticlesFixture',
+            $fixtures['plugin.Company/TestPluginThree.articles']
```
Comment (dereuromark) [inline tests/TestCase/TestSuite/FixtureManagerTest.php]: shouldnt that be uppercase `Articles`?

Comment (markstory) [inline tests/TestCase/TestSuite/FixtureManagerTest.php]: Normally you don't need to camelcase fixture names. e.g. `core.articles`, `app.bookmarks`, `plugin.bake.bake_articles` are all valid uses.

Comment (dereuromark) [inline tests/TestCase/TestSuite/FixtureManagerTest.php]: That's true, but I would rather stay consistent for both. Both snake_case or both CamelCase I guess. Nitpick, though.

**Comment by jadb** [comment]
@markstory I'll fix the tests - seems to have forgot to commit the fixture file in test_app/Company/.... Am not on my PC right now but will do as soon as I get on it.

@dereuromark ADmad suggested doing it for plugin.Vendor/Package.fixture to keep it inline with the rest of the framework's way of calling a vendor namespaced plugin. Let me know which notation it should support.

**Comment by ADmad** [comment]
@jadb @dereuromark's concern was only regarding the casing of the naming parts in general. The plugin name notation is fine.

**Comment by dereuromark** [comment]
Personally, taking the 3.0 chances towards less (unnecessary) inflection calls, I would see the snake_care versions as deprecated and would only - consistently - use the CamelCase one througout all files. Not sure what the general opinion on this is, though.

**Comment by ADmad** [comment]
:+1: to using CamelCase. Saves unnecessary inflection. (That's beyond the scope of this PR though)

**Comment by markstory** [comment]
I don't think the inflections are a deal breaker for me. I'm pretty sure you can use CamelBacked if you want and things will 'just work'.

**Comment by markstory** [comment]
Thanks @jadb

**Pull request merged**